File: extensions/task-with-files/scripts/after-read-file.js
#!/usr/bin/env node
/**
 * @license
 * Copyright 2026 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

/**
 * AfterTool Hook: read_file 专用
 *
 * 在读取文件后，通过 context 提示 LLM 如果有用知识则更新 findings.md
 * 仅在有进行中的任务时输出
 *
 * 通信方式：
 * - stdout: JSON 格式输出 (decision, context)
 * - stderr: 调试日志
 *
 * Note: Always exits with code 0
 */

import { readFileSync, existsSync } from 'fs';
import { join } from 'path';

const workspacePath = process.env.WORKSPACE_PATH || process.cwd();
const workingDirBase = join(process.cwd(), '.agent_working_dir');

const clean = (str) => str ? str.trim().replace(/^["']|["']$/g, '') : '';
const cleanWorkspacePath = clean(workspacePath);

// Get current task directory
const currentTaskFile = join(workingDirBase, 'current_task.json');

/**
 * 输出 JSON 到 stdout（必须保持纯净）
 */
function output(data) {
  console.log(JSON.stringify(data, null, 2));
}

/**
 * 调试日志输出到 stderr
 */
function logDebug(message) {
  console.error(`[after-read-file] ${message}`);
}

// 如果没有进行中的任务，输出 allow 并退出
if (!existsSync(currentTaskFile)) {
  logDebug('No current_task.json, skipping');
  output({ decision: 'allow' });
  process.exit(0);
}

try {
  const currentTaskData = JSON.parse(readFileSync(currentTaskFile, 'utf-8'));

  // 如果没有当前任务，输出 allow 并退出
  if (!currentTaskData.current) {
    logDebug('No active task, skipping');
    output({ decision: 'allow' });
    process.exit(0);
  }

  const taskDir = currentTaskData.current;
  const findingsFile = join(taskDir, 'findings.md');
  const findingsRelativePath = findingsFile.replace(cleanWorkspacePath + '/', '');

  // 通过 context 添加提示
  const context = `
📄 文件已读取

💡 如果获得了以下类型的信息，请更新 findings.md:
- 项目架构或目录结构
- 关键配置或依赖关系
- 重要的代码逻辑或算法
- 用户偏好或项目规范
- 技术决策或设计模式

📝 findings.md: ${findingsRelativePath}
`.trim();

  output({
    decision: 'allow',
    context: context
  });

} catch (err) {
  logDebug(`Error: ${err.message}`);
  output({ decision: 'allow' });
}

process.exit(0);

File: extensions/task-with-files/scripts/after_read_file.py
#!/usr/bin/env python3
import json
import os
import sys
from pathlib import Path


def get_current_task(workspace_path: str):
    if not workspace_path:
        return None

    current_task_file = (
        Path(workspace_path) / ".agent_working_dir" / "current_task.json"
    )

    if not current_task_file.exists():
        return None

    try:
        with open(current_task_file, "r") as f:
            data = json.load(f)
        return data.get("current")
    except:
        return None


def main():
    workspace_path = os.environ.get("WORKSPACE_PATH", "")

    try:
        input_data = json.load(sys.stdin)
    except json.JSONDecodeError:
        print(json.dumps({"decision": "allow"}), flush=True)
        return

    current_task = get_current_task(workspace_path)

    if not current_task:
        print(json.dumps({"decision": "allow"}), flush=True)
        return

    tool_input = input_data.get("tool_input", {})
    file_path = tool_input.get("file_path", "")

    if file_path:
        try:
            ws = Path(workspace_path)
            rel_path = Path(file_path).relative_to(ws)
        except ValueError:
            rel_path = Path(file_path)

        additional_context = f"""

---

**[task-with-files] File Read: {rel_path}**

If this contains useful information for your current task, consider adding a summary to **findings.md**.
"""
    else:
        additional_context = """

---

**[task-with-files] You just read a file.**

If this contains useful information for your current task, consider adding a summary to **findings.md**.
"""

    print(
        json.dumps(
            {
                "decision": "allow",
                "hookSpecificOutput": {"additionalContext": additional_context},
            }
        ),
        flush=True,
    )


if __name__ == "__main__":
    main()
